addons/misra_9.py

```diff
@@ -318,8 +318,9 @@ def parseInitializer(self, root, token):
                         if self.ed.parent != self.root:
                             # Check if token is correct value type for self.root.children[?]
                             child = self.root.getChildByValueElement(self.ed)
-                            if child.elementType != 'record' or self.token.valueType.type != 'record' or child.valueType.typeScope != self.token.valueType.typeScope:
-                                self.root.markStuctureViolation(self.token)
+                            if hasattr(self.token.valueType, 'type'):
```

Comment:
can you clarify this. It sounds very strange if valueType does not have a `type`.. how can that be? Can you tell me what this outputs?

    print(self.token.valueType)

Reply:
I would expect that valueType is either None or a fully valid `class ValueType` object defined in cppcheckdata.py.

Reply:
```
                            child = self.root.getChildByValueElement(self.ed)
                            if hasattr(self.token.valueType, 'type'):
                                if child.elementType != 'record' or self.token.valueType.type != 'record' or child.valueType.typeScope != self.token.valueType.typeScope:
                                    self.root.markStuctureViolation(self.token)
                            else:
                                print(self.token.valueType)

```
prints None on the screen.

Reply:
traceback.print_exc() gave me this:

```
  File "/./cppcheck/addons/misra.py", line 2040, in misra_9_5
    misra_9.misra_9_x(self, data, 905, rawTokens)
  File "/./cppcheck/addons/misra_9.py", line 416, in misra_9_x
    parser.parseInitializer(ed, eq.astOperand2)
  File "/./cppcheck/addons/misra_9.py", line 322, in parseInitializer
    if child.elementType != 'record' or self.token.valueType.type != 'record' or child.valueType.typeScope != self.token.valueType.typeScope:
AttributeError: 'NoneType' object has no attribute 'type'

```

I did not manage to isolate the exact piece of code in my project that is creating the issue. But I know what cpp-file is creating the issue, so I could try to isolate it.